main.py

```
@@ -97,6 +98,22 @@ def run_once(dry_run: bool = False) -> bool:
+    error_type: str,
+    error_message: str,
+    traceback_str: str | None = None,
+) -> None:
+    """Best-effort failure notification — never raises."""
+    try:
+        send_failure_notification(
+            error_type=error_type,
+            error_message=error_message,
+            traceback_str=traceback_str,
+        )
+    except Exception as notify_exc:
```

Ruff (0.15.2)

[error] Local variable `notify_exc` is assigned to but never used

Remove assignment to unused variable `notify_exc`

(F841)